From 0aace1a0b90f67356876c6d3fcdf023d6cefcaf3 Mon Sep 17 00:00:00 2001
From: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
Date: Thu, 27 Jan 2022 10:28:29 +0300
Subject: [PATCH 115/249] pstree: when updating sid for shell job also update
 matching pgid

If we replace old_sid with current_sid we should also do same
replacement for matching pgid (=old_sid).

Reported in CRIU gitter by Younes Manton (@ymanton)

Signed-off-by: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
---
 criu/pstree.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/criu/pstree.c b/criu/pstree.c
index d5080e515..d29de730a 100644
--- a/criu/pstree.c
+++ b/criu/pstree.c
@@ -382,6 +382,9 @@ static int prepare_pstree_for_shell_job(pid_t pid)
 		for_each_pstree_item(pi) {
 			if (pi->sid == old_sid)
 				pi->sid = current_sid;
+
+			if (pi->pgid == old_sid)
+				pi->pgid = current_sid;
 		}
 
 		if (lookup_create_item(current_sid) == NULL)
-- 
2.35.1

